src/reinforce.jl

```diff
@@ -111,21 +115,23 @@ function ChainRulesCore.rrule(
         map(_dist_logdensity_grad_partial, xs)
     end
 
+    ys_with_baseline = (variance_reduction && nb_samples > 1) ? ys .- y : ys
```

Comment:
This fails when elements of ys (and y) are not scalars. That's why I wrote something like that:
```julia
ys_with_baseline = map(ys) do yi
    yi .- y
end
```
Don't know if it can be any simpler